File: java/greedy/Jump_Game.java
public class Solution {
    public boolean canJump(int[] A) {
        int maxCover = 0;  
		    for(int start = 0; start<=maxCover && start<A.length;start++){  
		        if(A[start]+start > maxCover)  
		            maxCover = A[start]+start;  
		        if(maxCover >=A.length-1)  
		            return true;  
		    }  
		
		return false;	
    }
}

File: python/greedy/jump_game.py
class Solution:
    # @param A, a list of integers
    # @return a boolean
    def canJump(self, A):
        max_can_jump = 0
        for i, a in enumerate(A):
            if(max_can_jump < i):
                return False
            # update max_can_jump
            max_can_jump = max(max_can_jump, i + a)
        return True
